subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -1127,13 +1130,33 @@ public long getForkEvery() {
      * By default, Gradle automatically uses a separate JVM when executing tests, so changing this property is usually not necessary.
      * </p>
      *
-     * @param forkEvery The maximum number of test classes. Use null or 0 to specify no maximum.
+     * @param forkEvery The maximum number of test classes. Use 0 to specify no maximum.
+     * @since 8.1
      */
-    public void setForkEvery(@Nullable Long forkEvery) {
-        if (forkEvery != null && forkEvery < 0) {
+    public void setForkEvery(long forkEvery) {
+        if (forkEvery < 0) {
             throw new IllegalArgumentException("Cannot set forkEvery to a value less than 0.");
         }
-        this.forkEvery = forkEvery == null ? 0 : forkEvery;
+        this.forkEvery = forkEvery;
+    }
+
+    /**
+     * Sets the maximum number of test classes to execute in a forked test process.
+     * <p>
+     * By default, Gradle automatically uses a separate JVM when executing tests, so changing this property is usually not necessary.
+     * </p>
+     *
+     * @param forkEvery The maximum number of test classes. Use null or 0 to specify no maximum.
+     * @deprecated use the setter that takes a non-null long instead
+     */
+    @Deprecated
+    public void setForkEvery(@Nullable Long forkEvery) {
+        DeprecationLogger.deprecateInvocation("Test.setForkEvery(null)")
+            .replaceWith("Test.setForkEvery(0)")
+            .willBeRemovedInGradle9()
+            .withDslReference(Test.class, "forkEvery")
+            .nagUser();
```

Comment:
It looks like there are two deprecations here.
1. Calling the method with a Long: `setForkEvery(Long.valueOf(0))`
2. Calling the method with null: `setForkEvery(null)`

I think it makes sense to emit a deprecation for (2). Are there cases where (1) could be triggered where we wouldn't want to deprecate?

The current deprecation is for (2) only and it reads in an awkward way since we usually use this to refer users to a differently named method:
> Using method Test.setForkEvery(null) has been deprecated. This is scheduled to be removed in Gradle 9.0. Please use the Test.setForkEvery(0) method instead. 

Could we do...
1. `DeprecationLogger.deprecateBehavior` -- setting `forkEvery` with null. This is only emitted when forkEvery == null.
2. `DeprecationLogger.deprecateMethod` -- calling `setForkEvery` with a `Long`

Reply:
Very good point, thanks!
This is done